osu.Game/OsuGameBase.cs

```diff
@@ -281,11 +282,23 @@ List<ScoreInfo> getBeatmapScores(BeatmapSetInfo set)
             // we may want to revisit this if users notice or complain about the difference (consider this a bit of a trial).
             Audio.Tracks.AddAdjustment(AdjustableProperty.Volume, globalTrackVolumeAdjust);
 
-            Beatmap = new NonNullableBindable<WorkingBeatmap>(defaultBeatmap);
+            Beatmap = new NonNullableBindable<IWorkingBeatmap>(defaultBeatmap);
 
-            dependencies.CacheAs<IBindable<WorkingBeatmap>>(Beatmap);
+            dependencies.CacheAs<IBindable<IWorkingBeatmap>>(Beatmap);
             dependencies.CacheAs(Beatmap);
 
+            // this is kept around for EF use cases which eventually need to be removed or replaced.
+            typedBeatmap = new NonNullableBindable<WorkingBeatmap>(defaultBeatmap);
+            dependencies.CacheAs<IBindable<WorkingBeatmap>>(typedBeatmap);
+            dependencies.CacheAs(typedBeatmap);
+            typedBeatmap.BindValueChanged(b => Beatmap.Value = b.NewValue);
+            Beatmap.BindValueChanged(b =>
+            {
+                // handle the case of leasing. should always be a feedback event anyway.
```

Comment:
So the public interface-typed `Beatmap` to be considered the "leader"/"original" bindable here, and the EF-typed one is the "follower"/"copy"? I could be fine with that, but it's not super clear - just want to confirm my understanding.

Also if the event is supposed to always be a `Beatmap -> typedBeatmap -> Beatmap` feedback, then maybe a `Beatmap.Value == b.NewValue` check here is safer? If it isn't ever a feedback it would fail loudly rather than silently desync or something.

Reply:
I'm not actually sure which is the leader. I guess based on the order of the two bindable in this file your assumption could be correct.

The conditional you suggest does seem much safer, will give it a try.

Reply:
On looking again I don't think there is a "leader" here, both bindables need to be kept consistent with each other. Which is a bit of an issue, since - for instance - `OsuScreen` resolves and leases a `Bindable<WorkingBeatmap>`, which would be the `typedBeatmap` here. But because the lease would not affect the interface-typed bindable, since they're not bound to each other wrt disabled state, there is a possibility of a component which resolves `IBindable<IWorkingBeatmap>` trying to mutate it and causing a crash here.

I imagine a crash like this will not be possible given current code (because the current single bindable lease was making sure that someone would have seen a crash already on `master`), but when going forward with more changes it may become a possibility. So I think for full safety *both* `Value` and `Disabled` should be mirrored for both bindables here...?